src/torcontrol.cpp

```diff
@@ -532,8 +532,9 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
         }
 
         // Finally - now create the service
-        if (private_key.empty()) // No private key, generate one
-            private_key = "NEW:RSA1024"; // Explicitly request RSA1024 - see issue #9214
+        if (private_key.empty()) { // No private key, generate one
+            private_key = "NEW:ED25519-V3"; // Explicitly request key type - see issue #9214
+        }
```

Comment:
bb01c77865a565764e4e2f86265afef45506d7d7

The comment about #9214 seems outdated now:
```suggestion
        if (private_key.empty()) {
            // No private key, generate one.
            private_key = "NEW:ED25519-V3";
        }
```

Reply:
The comment is still relevant - we want to explicitly request the key type in order to avoid problems like #9214 in the future.

In other words - we don't want somebody to change this to `NEW:BEST` because if/when Tor upgrades to v4 then we may have #9214 again.

Reply:
Correct.